express/kfp_utils.py

```diff
@@ -1,6 +1,16 @@
 """General kfp helpers"""
+import os
 import ast
 import logging
+import yaml
+from typing import Callable
+
+import kfp
```

Comment:
This adds kfp as a dependency to express, which is a heavy dependency that's not needed for components. I would like to challenge if we really need it. And if we do, we should make it a soft dependency (can be done in a separate PR as this needs to be done for other dependencies as well).

Reply:
@NielsRogge would like your view on this as well since we've been discussing this before.

Reply:
We can create a `is_kfp_available` function to check whether kfp is installed, and throw an error if it it's not. This way, we can handle soft dependencies

Reply:
I'll implement this in a separate PR. Any pointers, examples on how this is handled in other repos?

Reply:
There's different ways to tackle this.
How we do it at Connexion: https://github.com/spec-first/connexion/pull/1668
HuggingFace does it differently, you can have a look there as well.

Reply:
Thanks, i'll check them out